File: javascript/leetcode/majority-element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    var seen = {}
    if (nums.length == 1) { return nums[0] }
    var majorityCount = ~~(nums.length / 2)
    for (var i = 0; i < nums.length; i++) {
        if (seen[nums[i]]) {
            seen[nums[i]]++
            if (seen[nums[i]] > majorityCount) {return nums[i]}
        } else { seen[nums[i]] = 1 }
    }
};

File: ruby/leetcode/majority-element.rb
# @param {Integer[]} nums
# @return {Integer}
def majority_element(nums)
    seen = {}
    if nums.size == 1
        return nums[0]
    end
    majority_count = nums.size / 2
    nums.each do |n|
        if seen.include? n
            seen[n] += 1
            if seen[n] > majority_count
                return n
            end
        else
            seen[n] = 1
        end
    end
end
